internal/resource/config/token_processor_to_token_generator_mapping_resource.go

```diff
@@ -0,0 +1,334 @@
+	internaljson "github.com/pingidentity/terraform-provider-pingfederate/internal/json"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/common/attributecontractfulfillment"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/common/attributesources"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/common/issuancecriteria"
+	internaltypes "github.com/pingidentity/terraform-provider-pingfederate/internal/types"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &tokenProcessorToTokenGeneratorMappingsResource{}
+	_ resource.ResourceWithConfigure   = &tokenProcessorToTokenGeneratorMappingsResource{}
+	_ resource.ResourceWithImportState = &tokenProcessorToTokenGeneratorMappingsResource{}
+)
+
+// TokenProcessorToTokenGeneratorMappingResource is a helper function to simplify the provider implementation.
+func TokenProcessorToTokenGeneratorMappingResource() resource.Resource {
+	return &tokenProcessorToTokenGeneratorMappingsResource{}
+}
+
+// tokenProcessorToTokenGeneratorMappingsResource is the resource implementation.
+type tokenProcessorToTokenGeneratorMappingsResource struct {
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+type tokenProcessorToTokenGeneratorMappingsResourceModel struct {
+	AttributeSources                 types.List   `tfsdk:"attribute_sources"`
+	AttributeContractFulfillment     types.Map    `tfsdk:"attribute_contract_fulfillment"`
+	IssuanceCriteria                 types.Object `tfsdk:"issuance_criteria"`
+	SourceId                         types.String `tfsdk:"source_id"`
+	TargetId                         types.String `tfsdk:"target_id"`
+	Id                               types.String `tfsdk:"id"`
+	CustomId                         types.String `tfsdk:"custom_id"`
+	DefaultTargetResource            types.String `tfsdk:"default_target_resource"`
+	LicenseConnectionGroupAssignment types.String `tfsdk:"license_connection_group_assignment"`
+}
+
+// GetSchema defines the schema for the resource.
+func (r *tokenProcessorToTokenGeneratorMappingsResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	schema := schema.Schema{
+		Description: "Manages Token Processor To Token Generator Mappings",
+		Attributes: map[string]schema.Attribute{
+			"custom_id": schema.StringAttribute{
+				Description: "The ID of the token processor to token generator mapping. The ID cannot be modified once the instance is created. Note: Ignored when specifying a connection's adapter override.",
+				Computed:    true,
+				Optional:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"attribute_contract_fulfillment": attributecontractfulfillment.AttributeContractFulfillmentSchema(true),
```

Comment:
This common stuff is really great. Really shortens the resource files and makes creating them easier.